cpp/models/abm/testing_strategy.h

```diff
@@ -0,0 +1,249 @@
+     * @param start_date starting date of the scheme
+     * @param end_date ending date of the scheme
+     * @param probability probability of the test to be performed if a testing rule applies
+     * @param test_type the type of test to be performed
+     */
+    TestingScheme(const std::vector<TestingCriteria>& testing_criteria, TimeSpan minimal_time_since_last_test,
+                  TimePoint start_date, TimePoint end_date, double probability, const GenericTest& test_type);
+
+    /**
+     * Compares two testing schemes for equality.
+     */
+    bool operator==(const TestingScheme& other) const
+    {
+        return this->m_testing_criteria == other.m_testing_criteria &&
+               this->m_minimal_time_since_last_test == other.m_minimal_time_since_last_test &&
+               this->m_start_date == other.m_start_date && this->m_end_date == other.m_end_date &&
+               this->m_probability == other.m_probability &&
+               this->m_test_type.get_default().sensitivity == other.m_test_type.get_default().sensitivity &&
+               this->m_test_type.get_default().specificity == other.m_test_type.get_default().specificity;
+        //To be adjusted and also TestType should be static.
+    }
+
+    /**
+     * add a testing criteria to the set of age groups that are checked for testing
+     * @param criteria testing criteria to be added
+     */
+    void add_testing_criteria(const TestingCriteria criteria);
+
+    /**
+     * remove a testing criteria from the set of age groups that are checked for testing
+     * @param criteria testing criteria to be removed
+     */
+    void remove_testing_criteria(const TestingCriteria criteria);
+
+    /**
+     * @return activity status of the scheme
+     */
+    bool is_active() const;
+
+    /**
+     * checks if the scheme is active at a given time and updates activity status
+     * @param t time to be updated at
+     */
+    void update_activity_status(const TimePoint t);
+
+    /**
+     * runs the testing scheme and tests a person if necessary
+     * @return if the person is allowed to enter the location
+     */
+    bool run_scheme(Person& person, const Location& location) const;
+
+private:
+    std::vector<TestingCriteria> m_testing_criteria;
+    TimeSpan m_minimal_time_since_last_test;
+    double m_probability;
+    TimePoint m_start_date;
+    TimePoint m_end_date;
+    GenericTest m_test_type;
+    bool m_is_active = false;
+};
+
+class TestingStrategy
+{
+public:
+    /**
+     * Create a testing strategy.
+     * @param testing_schemes vector of testing schemes that are checked for testing
+     */
+    TestingStrategy() = default;
+    explicit TestingStrategy(const std::vector<TestingScheme>& testing_schemes);
+    /**
+     * Compares two testing strategies for equality.
+     * Compare references. Still possible to clone strategies.
```

Comment:
remove?

Reply:
I removed the whole comparison operator, since I don't think we need it. I also adjusted some of the other comments of the other comparison operators.